client/python/gradio_client/utils.py

```diff
@@ -841,20 +841,36 @@ def set_space_timeout(
 ########################
 
 
-def synchronize_async(func: Callable, *args, **kwargs) -> Any:
+def synchronize_async(func, *args, **kwargs) -> Any:
```

Comment:
I benchmarked performance to see if this change would degrade, but it does not. cc @aliabid94 

<img width="657" alt="image" src="https://github.com/gradio-app/gradio/assets/1778297/d65e6573-2f65-45f4-b29d-e5dea2316009">